data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilter.java

```diff
@@ -938,6 +940,17 @@ public void testParquetTypePromotion() {
     Assert.assertTrue("Should succeed with promoted schema", shouldRead);
   }
 
+  @Test
+  public void testTransformFilter() {
+    Assumptions.assumeThat(format).isEqualTo(FileFormat.PARQUET);
+    boolean shouldRead =
+        new ParquetMetricsRowGroupFilter(SCHEMA, equal(truncate("required", 2), "some_value"), true)
+            .shouldRead(parquetSchema, rowGroupMetadata);
+    Assumptions.assumeThat(shouldRead)
```

Comment:
I think you want to use an assertion rather than an assumption? If you use an assumption the test will be skipped if it fails, instead of failing.

Reply:
Indeed, fixed.